src/deephaven/plugin/__init__.py

```diff
@@ -12,14 +16,30 @@ class Plugin(abc.ABC):
 
 
 class Registration(abc.ABC):
+    """
+    Registration types should be set as the registration_cls for deephaven.plugin entrypoints for their package to
+    ensure that they are all run on server startup.
+    """
     class Callback(abc.ABC):
         @abc.abstractmethod
         def register(self, plugin: Union[Plugin, Type[Plugin]]) -> None:
+            """
+            Registers a given plugin type for use in the Deephaven server. Should be called from from a Registration's
+            register_into method, so that it is available when the server expects it.
+            :param plugin: the plugin or plugin type to register on the server
+            :return:
+            """
             pass
```

Comment:
Inner classes are almost never done in Python.  @jmao-denver should give his thoughts.

Reply:
As long as we're already breaking compatibility, I don't see a problem with moving it.